llama_index/indices/vector_store/retrievers.py

```diff
@@ -47,6 +46,9 @@ def __init__(
         self._vector_store_query_mode = VectorStoreQueryMode(vector_store_query_mode)
         self._alpha = alpha
         self._doc_ids = doc_ids
+        self._filters = filters
+
+        self._kwargs: Dict[str, Any] = kwargs.get("retriever_kwargs", {})
```

Comment:
This feels a little weird. Should retriever kwargs be an actual kwarg in the init? I also don't see `retriever_kwargs` used anywhere 👀

Reply:
Ya agreed it's kinda weird. 

I started with just directly passing the kwargs through, then I remembered that we kinda abuse kwargs, e.g. when we do index.as_query_engine(...), we pass through all kwargs to both the retriever and the response synthesizer.

So the source of the problem is there.